kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/appliers/tasks/VariantTasks.kt

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.gradle.plugin.appliers.tasks
+
+import kotlinx.kover.gradle.plugin.appliers.artifacts.AbstractVariantArtifacts
+import kotlinx.kover.gradle.plugin.commons.*
+import kotlinx.kover.gradle.plugin.dsl.KoverVerifyBound
+import kotlinx.kover.gradle.plugin.dsl.internal.KoverReportFiltersConfigImpl
+import kotlinx.kover.gradle.plugin.dsl.internal.KoverReportVariantConfigImpl
+import kotlinx.kover.gradle.plugin.dsl.internal.KoverVerifyRuleImpl
+import kotlinx.kover.gradle.plugin.tasks.reports.*
+import kotlinx.kover.gradle.plugin.tasks.services.KoverPrintLogTask
+import kotlinx.kover.gradle.plugin.tools.CoverageTool
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.kotlin.dsl.register
+import org.gradle.language.base.plugins.LifecycleBasePlugin
+
+
+internal class VariantTasks(
+    private val project: Project,
+    private val variantName: String,
+    private val type: ReportVariantType,
+    private val toolProvider: Provider<CoverageTool>,
+    private val config: KoverReportVariantConfigImpl,
+    private val reporterConfiguration: Configuration
+) {
+    private val htmlTask: TaskProvider<KoverHtmlTask>
+    private val xmlTask: TaskProvider<KoverXmlTask>
+    private val binTask: TaskProvider<KoverBinaryTask>
+    private val verifyTask: TaskProvider<KoverVerifyTask>
+    private val logTask: TaskProvider<KoverFormatCoverageTask>
+
+    init {
+        htmlTask = project.tasks.createReportTask<KoverHtmlTask>(
+            htmlReportTaskName(variantName),
+            "Task to generate HTML coverage report for ${variantSuffix()}"
+        )
+        xmlTask = project.tasks.createReportTask<KoverXmlTask>(
+            xmlReportTaskName(variantName),
+            "Task to generate XML coverage report for ${variantSuffix()}"
+        )
+        binTask = project.tasks.createReportTask<KoverBinaryTask>(
+            binaryReportTaskName(variantName),
+            "Task to generate binary coverage report in IntelliJ format for ${variantSuffix()}"
+        )
+
+        verifyTask = project.tasks.createReportTask<KoverVerifyTask>(
+            verifyTaskName(variantName),
+            "Task to validate coverage bounding rules for ${variantSuffix()}"
+        )
+        logTask = project.tasks.createReportTask<KoverFormatCoverageTask>(
+            logTaskName(variantName),
+            "Task to print coverage to log for ${variantSuffix()}"
+        )
+        val printCoverageTask = project.tasks.register<KoverPrintLogTask>(printLogTaskName(variantName))
+
+        val runOnCheck = mutableListOf<Any>()
+
+        htmlTask.configure {
+            onlyIf { printPath() }
```

Comment:
> It looks like printPath always returns true

OK, it was just to avoid writing
```
onlyIf { printPath(); true }
```
:)

Reply:
> And also logs something. Should it be done on configuration stage?

This is specifically located here so that the path to the file is printed when the task is executed.
`onlyIf` is the only suitable action that is always executed, regardless of whether the task is cached or its actual execution is taking place